src/main/java/com/google/sps/youtube/YoutubeInfoScraper.java

```diff
@@ -0,0 +1,74 @@
+
+package com.google.sps.youtube;
+
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.youtube.YouTube;
+import com.google.api.services.youtube.model.ChannelListResponse;
+import com.google.api.services.youtube.YouTubeRequestInitializer;
+import com.google.api.services.youtube.model.Channel;
+import com.google.api.services.youtube.model.ChannelContentDetails;
+import com.google.api.services.youtube.model.ChannelContentDetails.RelatedPlaylists;
+import java.util.List;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import com.google.sps.youtube.YoutubeResponse;
+import java.util.Properties; 
+import java.io.FileInputStream;
+
+
+public class YoutubeInfoScraper implements YoutubeResponse {
+
+  // TODO: Add seperate file to hold API Key  
+  private String API_KEY = "";
+  private static final String APPLICATION_NAME = "promotube";
+
+  private YouTube youtube;
+
+  private YoutubeInfoScraper(YouTube youtube){
+      this.youtube = youtube;
+  }
+
+  private YoutubeInfoScraper(){
+    youtube = new YouTube.Builder(new NetHttpTransport(),
```

Comment:
even better is to have your constructors call each other like this:

this(new YouTube.Builder(new NetHttpTransport(),
        JacksonFactory.getDefaultInstance(),
        null)
        .setApplicationName(APPLICATION_NAME)
        .setYouTubeRequestInitializer(new YouTubeRequestInitializer(API_KEY))
        .build())

There's probably a best practice out there to do this.... if I had time to do some google searching.

Reply:
Done.  I was trying to find a way to have default parameters in Java.  I think this is as close as I can get.